File: routes/student.js
const express = require('express');
const { loginRequired } = require('../middleware/auth');
const { getUserById } = require('../models/user');

const router = express.Router();

function normalizeDate(value) {
  if (!value) return null;
  if (value instanceof Date) {
    return value.toISOString();
  }

  const asString = String(value);
  return asString.includes(' ') ? asString.replace(/\s/g, 'T') : asString;
}

router.get('/', loginRequired, async (req, res) => {
  try {
    const user = await getUserById(req.session.user_id);

    if (!user) {
      return res.status(404).json({ error: 'Student not found' });
    }

    return res.status(200).json({
      success: true,
      student: {
        id: user.id,
        username: user.username,
        email: user.email,
        created_at: normalizeDate(user.created_at)
      }
    });
  } catch (error) {
    return res.status(500).json({
      success: false,
      error: 'Failed to retrieve student information',
      details: error.message
    });
  }
});

module.exports = router;

File: routes/student.py
from flask import Blueprint, jsonify, session
from middleware.auth import login_required
from models.user import get_user_by_id

student_bp = Blueprint('student', __name__, url_prefix='/student')

@student_bp.route('/', methods=['GET'])
@login_required
def get_student_info():
    """
    Returns current student/user information as JSON
    """
    try:
        user_id = session.get('user_id')
        if not user_id:
            return jsonify({'error': 'User not authenticated'}), 401
        
        user = get_user_by_id(user_id)
        if not user:
            return jsonify({'error': 'Student not found'}), 404
        
        student_info = {
            'id': user['id'],
            'username': user['username'],
            'email': user['email'],
            'created_at': user['created_at'].isoformat() if user['created_at'] else None
        }
        
        return jsonify({
            'success': True,
            'student': student_info
        }), 200
    
    except Exception as e:
        return jsonify({
            'success': False,
            'error': 'Failed to retrieve student information',
            'details': str(e)
        }), 500
